### PR TITLE
sched.h: fix CPU_EQUAL(s1, s2) write error

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary

sched.h: fix CPU_EQUAL(s1, s2) write error
sched.h: add empty realize when NO CONFIG_SMP

## Impact

SMP

## Testing

QEMU

### CODE REVIEW

File: include/sched.h [hunk 1]
```diff
@@ -254,6 +254,9 @@ int    sched_getaffinity(pid_t pid, size_t cpusetsize, FAR cpu_set_t *mask);
 int    sched_cpucount(FAR const cpu_set_t *set);
 int    sched_getcpu(void);
 #else
+#  define sched_setaffinity(p, c, m) 0
+#  define sched_getaffinity(p, c, m) (*(m) |= (1 << (c)), 0)
```
Comment (xiaoxiang781216) [inline include/sched.h:258]:
```
((1 << (c)) - 1)
```

**Comment by masayuki2009** [comment]
>sched.h: fix CPU_EQUAL(s1, s2) write error

@GUIDINGLI 
Could you tell us how we can reproduce the error?

File: include/sched.h [hunk 2]
```diff
@@ -254,6 +254,9 @@ int    sched_getaffinity(pid_t pid, size_t cpusetsize, FAR cpu_set_t *mask);
 int    sched_cpucount(FAR const cpu_set_t *set);
 int    sched_getcpu(void);
 #else
+#  define sched_setaffinity(p, c, m) 0
+#  define sched_getaffinity(p, c, m) (*(m) |= ((1 << (c)) - 1), 0)
```
Comment (xiaoxiang781216) [inline include/sched.h:258]:
```suggestion
#  define sched_getaffinity(p, c, m) (*(m) = 1), 0)
```

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by GUIDINGLI** [comment]
> > sched.h: fix CPU_EQUAL(s1, s2) write error
> 
> @GUIDINGLI Could you tell us how we can reproduce the error?

This error is obvious, is a typo :
# define CPU_EQUAL(s1,s2) (*(s2) == *(s2))

And I found this error in a new patch which use CPU_EQUAL, and I will start a new PR to submit it, after this PR merged.

**Pull request merged**